plugins/plextor/qscan_cmd_plextor.cpp

```diff
@@ -0,0 +1,795 @@
+        *poe = ntoh32(dev->rd_buf+0x28) >> 1;
+    else
+        *poe = 0;
+    *pof = ntoh32(dev->rd_buf+0x10);
+
+#ifdef _debug_pi
+    printf(" * %4d * %4d\n", *pie, *poe);
+#endif
+    return 0;
+}
+#endif
+
+int scan_plextor::cmd_dvd_jb_block(cdvd_jb *data) {
+  for (int i = 0; i < 16; i++) {
+    int j = read_one_ecc_block(dev, dev->rd_buf, lba);
+    if (j == COMMAND_FAILED)
+      i = 16;
+    lba += 0x10;
+  }
+
+  cmd_jb_getdata(data);
+  if (dev->dev_ID > PLEXTOR_716) {
+    data->jitter = 3200 - 2 * data->jitter;
+  } else {
+    data->jitter = 3200 - (int)(2.4 * data->jitter);
+  }
+  return 0;
+  //	return (!(dev->rd_buf[2]));
+}
+
+int scan_plextor::cmd_cd_jb_block(cdvd_jb *data) {
+  int rsize = 15;
+  for (int i = 0; i < 5; i++) {
+    if (lba + 15 > dev->media.capacity)
+      rsize = dev->media.capacity - lba;
+
+    int j = read_cd(dev, dev->rd_buf, lba, rsize, 0xFA);
+    lba += rsize;
+    if (j == COMMAND_FAILED)
+      i++;
+  }
+
+  cmd_jb_getdata(data);
+  if (dev->dev_ID > PLEXTOR_716) {
+    data->jitter = 4800 - 2 * (data->jitter);
+  } else {
+    data->jitter = 3600 - (int)(2.4 * (data->jitter));
+  }
+  return 0;
+  //	return (!(dev->rd_buf[2]));
+}
+
+int scan_plextor::cmd_fete_getdata() {
+  dev->cmd[0] = PLEXTOR_FETE_READOUT; // 0xF5;
+  dev->cmd[3] = 0x0C;
+  dev->cmd[9] = 0xCE;
+  if ((dev->err = dev->cmd.transport(READ, dev->rd_buf, 0xCE))) {
+    sperror("PLEXTOR_FETE_READOUT", dev->err);
+    return dev->err;
+  }
+#ifdef _debug_fete
+  for (int i = 0; i < 0xCE; i++) {
+    if (!(i % 0x20))
+      printf("\n");
+    printf("%02X ", dev->rd_buf[i] & 0xFF);
+  }
+  printf("\n");
+#endif
+  return 0;
+}
+
+int scan_plextor::cmd_fete_block(cdvd_ft *data) {
+  int rdy = test_unit_ready(dev);
+  int offs = ((++fete_idx) << 1) + 8;
+  if (rdy != 0x20408) {
+    printf("test unit ready = %05X, return\n", rdy);
+    return -1;
+  }
+  //	if (cmd_fete_getdata()) return -1;
+  //	data->te=dev->rd_buf[offs];
+  //	data->fe=dev->rd_buf[offs+1];
+  data->te = 0;
+  data->fe = 0;
+  if (fete_idx > 99)
+    return 1;
+  lba = (fete_idx + 1) * fete_rsize;
+  if (lba > (dev->media.capacity_total - 1))
+    lba = dev->media.capacity_total - 1;
+
+  while ((data->fe == 0 || data->te == 0) && (fete_idx <= 99) &&
+         (rdy == 0x20408)) {
+    if (cmd_fete_getdata())
+      return -1;
+    data->te = dev->rd_buf[offs];
+    data->fe = dev->rd_buf[offs + 1];
+    if (data->fe == 0 || data->te == 0) {
+      rdy = test_unit_ready(dev);
+#ifdef _debug_fete
+      printf("test unit ready = %05X\n", rdy);
+#endif
+      msleep(10);
+    }
+  }
+  return 0;
+}
+
+int scan_plextor::build_TA_histogram_px716(unsigned char *response_data,
+                                           int *dest_pit, int *dest_land,
+                                           int len) {
+  int *dest[] = {dest_land, dest_pit};
+  int count = ntoh16(response_data + 2);
+  //	printf("PX-716 Histogram... %d\n",count);
+  int idx = 28;
+  int v, pit;
```

Comment:
Codacy found an issue: [The scope of the variable 'v' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6072101)